vnext/template/_gitattributes

```diff
@@ -0,0 +1,42 @@
+# Auto detect text files and perform LF normalization
+
+*       text=auto
+
+# C++ files should use CRLF.
+*.c   text eol=crlf
+*.cpp text eol=crlf
+*.h   text eol=crlf
+*.hpp text eol=crlf
+
+# Spec Codegen uses LF
+*Spec.g.h eol=lf
+
+# Force Visual Studio project files (mostly XML) to CRLF
+#   This helps avoid conflict which occurs when Xmarian/Mac contributors with AutoCRLF=input (or off)
+#   check in these files as CRLFs, and then Windows contributors with AutoCRLF=true end up checking then
+#   files in as LFs  (due to AutoCRLF=true normalizing CRLF->LF on check-in).
+
+*.sln text eol=crlf
+*.csproj text eol=crlf
```

Comment:
nit: I often search for msbuild files via `*.*proj`
Wild suggestion: Should we just use that wildcard, or should we list explicitly?